utils/s2n_blob.c

```diff
@@ -102,19 +102,22 @@ static const uint8_t hex_inverse[256] = {
 int s2n_hex_string_to_bytes(const char *str, struct s2n_blob *blob)
 {
     notnull_check(str);
-    notnull_check(blob);
+    PRECONDITION_POSIX(blob);
     uint32_t len = strlen(str);
     /* protects against overflows */
     gte_check(blob->size, len / 2);
     S2N_ERROR_IF(len % 2 != 0, S2N_ERR_INVALID_HEX);
 
     for (int i = 0; i < len; i += 2) {
+/* Avoids spurious alarms over signed to unsigned conversion (uint8_t)str[(signed long int)i]. */
```

Comment:
I would much rather fix these errors rather than ignore them. The `i` type should be a `size_t` rather than an `int`.